x/auth/legacy/legacytx/msg_service.go

```diff
@@ -0,0 +1,40 @@
+package legacytx
+
+import (
+	gocontext "context"
+	"fmt"
+
+	"google.golang.org/grpc"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// Invoke implements the grpc ClientConn.Invoke method. This is so that we can
+// use ADR-031 service `Msg`s with StdTxBuilder. Invoking this method will
+// **append** the service Msg into the TxBuilder's Msgs array.
+// TODO Full amino support still needs to be added as part of https://github.com/cosmos/cosmos-sdk/issues/7541.
```

Comment:
This PR was originally set up in #7541. It introduces a clunky UX to manage service Msgs with TxBuilder, see https://github.com/cosmos/cosmos-sdk/pull/7630#issuecomment-718315664. It should be ultimately replaced by a custom code generator https://github.com/cosmos/cosmos-sdk/issues/8270.

However, as I understand, #8270 is targetted for 0.42, whereas this PR might be useful for `x/fee_grant` and `x/authz` (see https://github.com/cosmos/cosmos-sdk/pull/8061#discussion_r555081173, https://github.com/cosmos/cosmos-sdk/pull/7629#discussion_r540880541, both use a temporary `ServiceMsgClientConn` struct for now).

Should we:

1. Merge this, and have a slightly clunky txBuilder for now?
2. Close this, use temporary `ServiceMsgClientConn` wherever we need service Msgs, and wait for #8270?

I think 1 would make the path to `x/fee_grant` and `x/authz` easier, and we could refactor `TxBuilder` as a whole when considering #8270.

Reply:
Talking offline with the Regen team, I'll close this PR.

- `TxBuilder` won't be able to handle service Msgs (until #8270)
- If we want to use service msgs, we use the temporary ServiceMsgClientConn.